src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -602,10 +602,14 @@ private OpenApiParameter GenerateParameterWithoutFilter(
                 In = location,
                 Required = isRequired,
                 Schema = schema,
-                Description = description
+                Description = description,
+                Style = GetParameterStyle(type, apiParameter.Source)
             };
         }
 
+        private static ParameterStyle? GetParameterStyle(Type type, BindingSource source) =>
+            source == BindingSource.Query && type == typeof(Dictionary<string, string>) ? ParameterStyle.DeepObject : null;
```

Comment:
I am not sure about the implementation, I decided to keep strict type equality to `Dictionary<string, string>`, but maybe in swagger makes sense to make any [FromQuery] IDictionary type (or type that implements IDictionary) to have "style": "deepObject"

Reply:
Other types we could consider checking instead could be:

- `IDictionary<string, string>`
- `IReadOnlyDictionary<string, string>`
- `IEnumerable<KeyValuePair<string, string>>`

Reply:
applied suggestion